### PR TITLE
test/testdrive: use materialized sources in more places

### PR DESCRIPTION
**Pull request by benesch**
This a) vastly decreases file descriptor usage, and b) speeds up
testdrive, because we don't create a billion independent Kafka sources.

### CODE REVIEW

**Comment by frankmcsherry** [comment]
Does this substantially change FD use for anything other than `multijoins.td`? All of the others seemed to use materialized views around their sources (and the PR tidies them for sure, but I'm trying to grok whether there is a difference to warn people about / fix).

**Comment by frankmcsherry** [comment]
And, I guess related, do we have any idea why four queries in `multijoins.td` would prompt this amount of FD use? Is the reduction more than 4x? I'm all for improving the tests, but it seems like we probably still have an issue to debug somewhere, either in the code or in our understanding of what resources are needed for a set of queries and views.

**Comment by benesch** [comment]
> Does this substantially change FD use for anything other than `multijoins.td`? All of the others seemed to use materialized views around their sources (and the PR tidies them for sure, but I'm trying to grok whether there is a difference to warn people about / fix).

Yeah, it's really just multijoins.

I'm still pretty convinced that there isn't a concerning issue here. rdkafka uses ~10 FDs per consumer. We now create two consumers per source—one for the timestamper and one for the dataflow worker. So that's 20 FDs per source right there. And then if you have a materialized view that joins three non-materialized sources, that's an easy 60 FDs.

File: test/testdrive/joins.td [hunk 1]
```diff
@@ -225,13 +221,3 @@ num name num mod
 <null> <null> 0 even
 <null> <null> 1 odd
 <null> <null> 2 even
-
-> CREATE VIEW test15 AS
```
Comment (wangandi) [inline test/testdrive/joins.td]: I am still in favor of having at least one test in testdrive that is a join between a materialized and an unmaterialized view/source.

Comment (benesch) [reply test/testdrive/joins.td]: Oh, I didn't realize that's what this was intended to be testing. You mean the fact that joining one materialized and one unmaterialized source results in an error?

Comment (benesch) [reply test/testdrive/joins.td]: Ok, restored this block and merging!

**Pull request merged**